fix: add controlled prop

felix-ico opened the pull request:
So using the same approach as in the dropdown fixes this [react/reactivity issue](https://github.com/telekom/scale/issues/805), I guess this would need to be added to some other components as well to avoid the same issue in the future (at least in text-area).

acstll commented on packages/components/src/components/text-field/text-field.tsx:
```diff
@@ -156,6 +168,11 @@ export class TextField {
   handleInput = (event: Event) => {
     const target = event.target as HTMLInputElement | null;
+    if (this.controlled) {
+      emitEvent(this, 'scaleChange', { value: target.value });
```
Is firing `scaleChange` here really needed? it's being fired again on line 178. Also, doesn't the re-render triggered by `forceUpdate` fire another change event?

Maybe it'd make sense to add a test to check whether the change event is not being fired too much?

felix-ico replied:
yup that was a mistake, removed and added a test

acstll replied:
amazing, thank you!

acstll reviewed (commented):
Also, should we also document this new `controlled` prop in Storybook? I think yes, but with a notice saying it's "experimental"… maybe?

felix-ico commented on packages/components/src/components/text-field/text-field.tsx:
```diff
@@ -96,6 +96,8 @@ export class TextField {
   @Prop() styles?: string;
 
+  /** (optional) Makes type `input` behave as a controlled component in React */
+  @Prop() controlled?: boolean = false;
```
so i replaced (optional) with (experimental) but I'm not sure if that is what you meant or to change the prop name to something like `experimental_controlled` ? Wasn't sure because in other components like the dropdown there is also a `controlled` prop but without any experimental prefix

acstll replied:
yes, haha, I was thinking about `experimental_controlled` — the old Dropdown is going to be deprecated.

Would it work with the snake_case? 🤔

acstll commented on packages/components/src/components/text-field/text-field.tsx:
```diff
@@ -95,7 +95,8 @@ export class TextField {
   /** (optional) Injected CSS styles */
   @Prop() styles?: string;
-
+  /** (experimental)) Makes type `input` behave as a controlled component in React */  
```
there's an extra ) in `(experimental))` 🤓

felix-ico replied:
> Would it work with the snake_case? 🤔

seems like it doesn't work, the linter complains. experimentalControlled/experimental-controlled works

acstll replied:
I'm not in love with `experimental-controlled` but I think I prefer it over just `controlled` (only because it's a React-only thing)

acstll commented:
I got confused with the many "conversations" I opened in the comments, haha, sorry 🙏

felix-ico commented:
@acstll all CI jobs are passing now, snake_case doesn't work so i changed it to experimental-controlled. 

I also added the prop to storybook, but there is no real documentation on why/how someone would use it. Should that be included in storybook as well? or maybe just in the readme?

acstll reviewed (approved):
🙈
